### ISSUE TITLE
Misspelled policy.yaml keys silently disable their rule: the #302 unknown-key warning is filtered out in release builds

### ISSUE THREAD

**Issue by igouss** [Dicklesworthstone/beads_rust#515]
## What happened

A misspelled key under `close_policy` or `workflow` in `.beads/policy.yaml` turns that rule off, and a release build of `br` prints nothing about it. The warning added for #302 exists, but it is emitted with `tracing::warn!`, and in release builds the default log filter is `error`, so the warning only appears with `-v`. `br doctor` does not report the unknown key either.

Repro in a fresh repository (release binary):

```text
$ git init -q . && br init --prefix demo
$ cat .beads/policy.yaml
close_policy:
  require_close_reasn: {enabled: true, min_length: 40}

$ br create "Demo" -t task --silent
demo-shs
$ br close demo-shs --reason ok
✓ Closed demo-shs: Demo (ok)
exit=0                                   # stderr empty: the typo disabled the rule silently

$ br -v close demo-jpy --reason ok 2>&1 | grep WARN
... WARN beads_rust::close_policy: policy.yaml contains 1 unknown field(s) under close_policy structs; these were ignored (beads_rust#302). Check for typos: close_policy.require_close_reasn ...

# control: the key spelled correctly
$ br close demo-19w --reason ok
Error: Policy violation for demo-19w: close_reason fails policy: minimum length is 40 chars (got 2)
exit=4
```

The same happens under `workflow`: with `strickt: true` instead of `strict: true`, `br create "x" -s bogus_status` succeeds with exit 0 and no output on stderr. With `strict: true` it is refused ("status 'bogus_status' is not permitted by the project workflow policy"). Again, the warning appears only with `-v`.

`br doctor --json` on the workspace with the misspelled key: its output does not contain `require_close_reasn` anywhere.

The capacity section handles the same mistake the opposite way: `in_review: {soft: 3, hardd: 5}` fails at load with exit 7 ("unknown field `hardd`, expected `soft` or `hard`"). So one policy file fails closed on a typo in one section and fails open, invisibly, in the others.

## Why it matters

`policy.yaml` is a set of guardrails. Most people and agents run `br` without `-v`, and in that normal case a typo leaves a rule that looks configured but isn't enforced. For example, a misspelled `forbid_self_close_after_in_progress` lets an agent close its own work, and nothing ever says so.

## Where

At commit `adebe3ff`:

- `src/close_policy.rs:2852`: the #302 detection emits `tracing::warn!(...)`.
- `src/logging.rs:85` (`default_filter`): release builds with verbosity 0 use `"error"`. The test at `src/logging.rs:143` asserts exactly that.

## Possible fixes (for illustration)

#302 deliberately turned unknown keys from a hard load failure (which blocked every close in the project) into a warning, so these suggestions keep the load non-fatal and only make the warning reachable:

- Print the unknown-key notice to stderr directly, not through `tracing`, so it shows at default verbosity. It could be printed once per invocation, and in `--json` mode kept on stderr so stdout stays parseable.
- Add a `br doctor` finding for unknown `policy.yaml` keys, so `br doctor --quick` in CI catches them.

## Version and platform

```text
$ br --version
br 0.6.0
```

Release build, commit `3b857195` (features: `self_update`). The relevant code is unchanged at `adebe3ff`. Linux 7.1.3 (Fedora 43), x86_64. Repro run on tmpfs.

## Selftest receipt

```text
$ RUST_LOG=error br doctor --selftest --json
ok: true (platform linux/x86_64, br_version 0.6.0)
```

## Incident bundle

Not applicable. This is a configuration-parsing and logging issue, not workspace state. The repro above is self-contained.

**Comment by igouss** [comment]
I think your decision for #302 "Design choice: Option A (relaxed parsing with named warnings)." was wrong, I think you should have strict validation.
